docs/contracts/v4/concepts/07-deployment.mdx

```diff
@@ -0,0 +1,141 @@
+
+1. **CREATE**: This is the regular way. Every time you create a contract using this, it gets a new, unique address (like a house getting a unique postal address). You can't fully predict this address in advance.
+
+2. **CREATE2**: This is a special way. Here, you use some ingredients (your address, a `salt` which is a unique number you choose, and the contract's code called `bytecode`) to create the contract. The magic of `CREATE2` is that if you use the same ingredients, you'll get the same contract address every time.
+
+So, why use `CREATE2`? Because sometimes you want to know exactly where your contract will be, even before you create it. Like in your hooks system, where certain addresses trigger certain actions, using `CREATE2` helps ensure the contract is deployed to the exact right address.
+
+Here's a small code to show how `CREATE2` works:
+```solidity
+bytes32 salt = keccak256(abi.encodePacked(someData));
+address predictedAddress = address(uint(keccak256(abi.encodePacked(
+    byte(0xff),
+    deployerAddress,
+    salt,
+    keccak256(bytecode)
+))));
+```
+This code predicts the address where a contract will be deployed using `CREATE2` before actually deploying it.
+
+# Deterministic Deployment Proxy
+Many developers use https://github.com/Arachnid/deterministic-deployment-proxy to deploy contracts to a specific
+address. The main feature of this project is the use of the Ethereum CREATE2 opcode, which allows for deterministic
+deployment of contracts. This means that using a specific set of inputs (bytecode hash and salt), the deployed
+contract will always end up at the same address across different networks.
+
+Deterministic deployment proxy uses two key techniques:
+1) The proxy contract is deployed using a one-time private key, so it will always have the same address no matter when
+or where it is deployed.
+2) The proxy contract uses the CREATE2 opcode to deploy the target contract. CREATE2 calculates the address of the
+deployed contract based on the init code and a salt parameter. By using the same init code (bytecode) and salt each
+time, the target contract will be deployed to the same address across networks.
+
+Most of the chains do have the deployment proxy at `0x4e59b44847b379578588920cA78FbF26c0B4956C`. See [here](https://github.com/Uniswap/v4-periphery/issues/59#issuecomment-1716379675)
+for more details.
+
+# Hook Deployment Code
+The https://github.com/uniswapfoundation/v4-template repository contains some helper utilities for deploying hooks.
+
+Here is the code for deploying the hooks using Deterministic Deployment Proxy which is deployed at `0x4e59b44847b379578588920cA78FbF26c0B4956C`:
+```solidity
+contract CounterScript is Script {
+    address constant CREATE2_DEPLOYER = address(0x4e59b44847b379578588920cA78FbF26c0B4956C);
+    address constant GOERLI_POOLMANAGER = address(0x3A9D48AB9751398BbFa63ad67599Bb04e4BdF98b);
```

Comment:
`GOERLI_POOLMANAGER` --> `POOLMANAGER`

also would set its address to a placeholder like `0xBEEF` or something